aoc-client/src/lib.rs

```diff
@@ -421,6 +421,77 @@ impl AocClient {
         Ok(())
     }
 
+    fn get_global_leaderboard_html(&self) -> AocResult<String> {
+        debug!("🦌 Fetching global leaderboard for {}", self.year);
+
+        let url = format!("https://adventofcode.com/{}/leaderboard", self.year);
+        let response = reqwest::blocking::get(url)?;
+        let contents = response.error_for_status()?.text()?;
```

Comment:
It might be a good idea to handle NOT FOUND which can only happen if the year is invalid (like it's done in `get_calendar_html`):

```
        if response.status() == StatusCode::NOT_FOUND {
            // A 404 reponse means the leaderboard for
            // the requested year is not yet available
            return Err(AocError::InvalidEventYear(self.year));
        }
```

Reply:
Makes sense! Added :)